src/components/Upload.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+
+const Upload = ({ setCurrentScript, titles, setTitles }) => {
+  const MAXFILESIZE = 50 * 1024 * 1024; // 50MB
+  const [file, setFile] = useState();
+  const [errMessage, setErrMessage] = useState({ message: '', display: 'none', color: 'red' });
+
+  const handleFileChange = (e) => {
+    e.preventDefault();
+    // check that the file size is less than 50 before allowing the upload
+    if (e.target.files) {
+      if (e.target.files[0].size > MAXFILESIZE) {
+        setErrMessage({ message: 'Selected file is too large. Max size is 50MB', display: 'block', color: 'red' });
+        setFile(null);
+      } else {
+        setFile(e.target.files[0]);
+        setErrMessage({ message: '', display: 'none' });
+      }
+    }
+  };
+
+  const handleUploadClick = () => {
+    if (!file) {
+      return;
+    }
+
+    // set the file as Form Data and send it to the server
+    const formData = new FormData();
+    formData.append('newScript', file);
+
+    fetch('/script', {
+      method: 'POST',
+      body: formData,
+    })
+      .then((response) => {
+        if (response.status === 409) {
+          throw new Error('script already exists');
+        }
+        return response.json();
+      })
+      .then((title) => {
+        if (title) {
+          // set the current script to the newly uploaded script
+          setCurrentScript(title);
+          setTitles([...titles, title]);
+          // reset error message
+          setErrMessage({ message: 'Successfully Uploaded', display: 'block', color: 'black' });
+          setFile(null);
+        }
+      })
+      .catch((err) => {
+        if (err.message === 'script already exists') {
+          setErrMessage({ message: 'That script title already exists.', display: 'block', color: 'red' });
+        } else {
+          setErrMessage({ message: 'File was not uploaded. Try again.', display: 'block', color: 'red' });
```

Comment:
Not much of an issue because this is contained within this one component, but it can be cleaner to represent the message and the display properties separately.  This keeps the presentation pieces all in once place and the content separate.   
For example:
```js
setErrMessage('Script already exists')
```
and
```jsx
{errMessage.length > 0 &&
    <p style={{color: red}}>{errMessage}</p>
}
```

See https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator
  
or if you need to support multiple styles:  
```js
setMessage({message: 'Script already exists', error: true})
setMessage({message: 'Script uploaded', error: false})
```
and  
```jsx
{errMessage.length > 0 &&
    <p style={error === true && {color: red}}>{errMessage}</p>
}
```

Reply:
Excellent suggestion. I had been feeling that wasn't very DRY but wasn't sure how to approach making it more concise. Thank you for the react doc link too, I had emulated someone else's component for the file inline conditional rendering and I didn't fully understand it until now.